acceptancetests/deploy_stack.py

```diff
@@ -592,11 +592,13 @@ def create_initial_model(self, upload_tools, series, boot_kwargs):
 
     def get_hosts(self):
         """Provide the controller host."""
-        host = get_machine_dns_name(
+        hosts = {}
+        host_0 = get_machine_dns_name(
             self.client.get_controller_client(), '0')
-        if host is None:
-            raise ValueError('Could not get machine 0 host')
-        return {'0': host}
+        if host_0 is not None:
+            # Only IaaS controller has machine now.
```

Comment:
can we add a check that this is a caas controller?

Reply:
as discussed, currently the controller type is not accessible from here.